types/index.d.ts

```diff
@@ -1,21 +1,84 @@
+/**
+ * A database transaction
+ */
 export type Transaction = Array<object>;
+/**
+ * The Schema is a way to to simplify relational queries at query time.
+ * Presently, it allows you to define relationships and uniqueness constraints.
+ */
 export type Schema = object;
+/**
+ * Homebase configuration.
+ */
 export type config = {schema?: Schema, initialData?: Transaction };
+/**
+ * Entities are JSON objects with extra features.
+ */
 export type Entity = {
-  get: (...params:string[]) => any;
+  /**
+   * Get a value from the `Entity` given its attribute(s).
+   * You can traverse arbitrarily deep relationships by adding nested attributes to the attributes array.
+   * @param attributes a string array of parameter names
+   */
+  get: (...attributes:string[]) => any;
 }
+/**
+ * Facts are richer key-value pairs that represent your data.
```

Comment:
"Datoms are the smallest unit of data in the database. They are key-value pairs with extra information like entity id, transaction id, and whether this key was added or deleted from the database."